File: SelectionSort.c
#include <stdio.h>

void InputArray (int array[] , int m) {
    for (int i=0; i<m; i++) {
        printf("Enter %d element: ",i+1);
        scanf("%d",&array[i]);
    }
}

void swap (int array[] , int a , int b) {
    int temp = array[a];
    array[a] = array[b];
    array[b] = temp;
}

void SelectionSort (int array[] , int m) {
    for (int i=0; i<m-1; i++) {
        int min = i;
        for (int j=i+1; j<m; j++) {
            if (array[j] < array[min]) {
                swap(array , min , j);
            }
        }
    }
}

void PrintArray (int array[] , int m) {
    SelectionSort(array , m);
    for (int i=0; i<m; i++) {
        printf("%d ",array[i]);
    }
}

void main () {
    int m;
    printf("Enter size of array: ");
    scanf("%d",&m);

    int array[m];

    InputArray(array , m);
    PrintArray(array , m);
}

File: SelectionSort.java
// find the shortest and swap

public class SelectionSort {
    public static void main(String[] args) {
        int arr[] = {7, 3, 2, 111, 5, 1};

        ss(arr, arr.length);
        
        for(int i=0; i<arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
    }

    public static void ss(int arr[], int n) {
        for(int i=0; i<n-1; i++) {
            int min = i;
            for(int j=i+1; j<n; j++) {
                if(arr[j] < arr[min]) {
                    min = j;
                }
            }
            swap(arr, i, min);
        }
    }

    public static void swap(int arr[], int x, int y) {
        int temp = arr[x];
        arr[x] = arr[y];
        arr[y] = temp;
    }
}

